plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -51,15 +52,18 @@
 
     private final StorageClass storageClass;
 
-    S3BlobStore(S3Service service, String clientName, String bucket, boolean serverSideEncryption,
-                ByteSizeValue bufferSize, String cannedACL, String storageClass) {
+    private final Tuple<RepositoryMetaData, Settings> settingsKey;
+
+    S3BlobStore(S3Service service, String bucket, boolean serverSideEncryption,
```

Comment:
This constructor could be simplified now, making all the setting extraction from the metadata happen here. I just left that out of this step because it requires some noisy changes in the tests.